pkg/ingester/ingester.go

```diff
@@ -577,6 +578,25 @@ func (i *Ingester) removeShutdownMarkerFile() {
 func (i *Ingester) loop() {
 	defer i.loopDone.Done()
 
+	// Delay the first flush operation by up to 0.8x the flush time period.
+	// This will ensure that multiple ingesters started at the same time do not
+	// flush at the same time. Flushing at the same time can cause concurrently
+	// writing the same chunk to object storage, which in AWS S3 leads to being
+	// rate limited.
+	jitter := rand.Int63n(int64(float64(i.cfg.FlushCheckPeriod.Nanoseconds()) * 0.8))
```

Comment:
isn't that too little? I feel like you'll still hit the race condition if you only sleep for 0.8 ns